src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -122,6 +122,17 @@ public function load(array $configs, ContainerBuilder $container)
             $twigFilesystemLoaderDefinition->addMethodCall('addPath', array($dir));
         }
 
+        if ($container->fileExists($dir = $container->getParameter('kernel.project_dir').'/templates', false)) {
+            $twigFilesystemLoaderDefinition->addMethodCall('addPath', array($dir));
```

Comment:
Registering automatically a directory to override bundles templates would require a configuration key and this can easily be managed from the TwigExtension, `templates/bundles` being then a default if that's really a concern ;).

For now this is also adding the new conventional `templates` folder (the `bundles` sub directory being a pending proposal), as it was done for `app/Ressources/views` a few lines above.

Anyone is still free to add its own paths, even use the new templates directory structure in symfony 2.8 if wanted. Also the addition happens only if the `templates` folder exists which should be the new best practice now, we used to assume that `app/Resources/views` and `app/Resources/*Bundle/views` exist as well.